src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -29,15 +30,17 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
             ConnectionControl = this;
 
             _connectionId = Interlocked.Increment(ref _lastConnectionId);
+
+            SocketInput = new SocketInput(Memory2);
+            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
+            _frame = new Frame(this);
         }
 
         public void Start()
         {
             Log.ConnectionStart(_connectionId);
 
-            SocketInput = new SocketInput(Memory);
-            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
-            _frame = new Frame(this);
+            Task.Run((Action)_frame.ProcessFraming);
```

Comment:
K, makes sense; then I'd just change ```Task.Run``` -> ```ThreadPool.QueueUserWorkItem``` as per snippet you are ignoring the Task results and catching all the Exceptions in ```ProcessFraming``` and ```Task.Run``` allocates more to allow for StackTrace unwinding in whatever awaits it; and eventually just calls ```ThreadPool.QueueUserWorkItem```